src/lib/map.c

```diff
@@ -1,2 +1,37 @@
 #include "lib/map.h"
 
+int map_init(struct map* m, uint64_t size) {
+	m->table = calloc(size, sizeof(struct map));
+	m->size = size;
+
+	/** Ensure map was allocated. */
+	if (m->table == NULL) return -1;
+	else return 0;
+}
+
+void map_free(struct map* m) {
+	free(m->table);
+}
+
+void map_put(struct map* m, void* key, void* value) {
```

Comment:
Your map_put function doesn't account for hash collisions and instead removes the previous key/value pair, making it inaccessible from the map, and a potential issue for memory management.

Reply:
I noticed that: fnv_1a will never collide however the modulus operation may make it collide. I'll rectify this soon.

Reply:
That's not necessarily the case, there is always the chance that two variables happen to have the same hash value. Hash collisions are probabilistic which isn't ideal if we end up using a map for something important. What this means for your code is that hash collisions should be resolved for arbitrarily large values of `m->size` (a size of 2^64-1 is equivalent to not using the modulus at all)

Reply:
Ensured collision will never occur.